scripts/linux/nestedAgentLock.sh

```diff
@@ -0,0 +1,243 @@
+#! /bin/bash
+
+###############################################################################
+# This script is used to "lock" agents for nested tests. When we lock an agent,
+# we alter the agent capability named "status" by appending the build id of
+# the current run.
+
+# Locking agents is not an atomic operation so this script has to account for
+# race conditions.
+#
+# We will attempt to lock 3 agents, then wait a bit for the booked agents
+# lock-state to become non-volatile. If all agents are still locked with the 
+# build id this script is using, then we know we can proceed safely. 
+#
+# If instead, another build somehow booked some of our previously-booked agents
+# after we did, we can unlock all the agents we still have booked. Then wait a
+# random duration to avoid multiple instances of the script thrashing on the
+# next booking attempt.
+###############################################################################
+
+set -e
+
+###############################################################################
+# Define Environment Variables
+###############################################################################
+
+SCRIPT_NAME=$(basename $0)
+
+POOL_ID=123
```

Comment:
Nit: Please add a comment regarding the name of the pool

Reply:
I made the change. We could also inject this from outside the script to keep it non-coupled from our test environment. Feel like this will never be run on any other pool though.

Reply:
I hear you. I just want to at lease make it clear on to whom we are targeting.
It maybe very very obvious at this point, but who knows 6 months from now if we have another "obvious" pool.

Reply:
> Feel like this will never be run on any other pool though.

Yeah I agree with you. This point was just regarding injecting to the script.